infra/cifuzz/cifuzz.py

```diff
@@ -319,25 +322,75 @@ def get_files_covered_by_target(latest_cov_info, target_name,
     logging.info('No files found in coverage report.')
     return None
 
-  # Cases like curl there is /src/curl and /src/curl_fuzzers/ are handled.
-  if not oss_fuzz_project_base.endswith('/'):
-    oss_fuzz_project_base += '/'
+  # Make sure cases like /src/curl and /src/curl/ are both handled.
+  norm_src_in_docker = os.path.normpath(src_in_docker)
+  if not norm_src_in_docker.endswith('/'):
+    norm_src_in_docker += '/'
 
   affected_file_list = []
   for file in coverage_per_file:
-    if not file['filename'].startswith(oss_fuzz_project_base):
+    norm_file_path = os.path.normpath(file['filename'])
+    if not norm_file_path.startswith(norm_src_in_docker):
       continue
     if not file['summary']['regions']['count']:
       # Don't consider a file affected if code in it is never executed.
       continue
 
-    relative_path = file['filename'].replace(oss_fuzz_project_base, '')
+    relative_path = file['filename'].replace(norm_src_in_docker, '')
     affected_file_list.append(relative_path)
   if not affected_file_list:
     return None
   return affected_file_list
 
 
+def remove_unaffected_fuzzers(project_name, out_dir, files_changed,
+                              src_in_docker):
+  """Removes all non affected fuzzers in the out directory.
+
+  Args:
+    project_name: The name of the relevant OSS-Fuzz project.
+    out_dir: The location of the fuzzer binaries.
+    files_changed: A list of files changed from origin.
+    src_in_docker: The location of the source dir in the docker image.
+  """
+  if not files_changed:
+    logging.info('No files changed from origin.')
+    return
+  fuzzer_paths = utils.get_fuzz_targets(out_dir)
+  if not fuzzer_paths:
+    logging.error('No fuzzers found in out dir.')
+    return
+
+  latest_cov_report_info = get_latest_cov_report_info(project_name)
+  if not latest_cov_report_info:
+    logging.error('Could not download latest coverage report.')
+    return
+  affected_fuzzers = []
+  for fuzzer in fuzzer_paths:
+    covered_files = get_files_covered_by_target(latest_cov_report_info,
+                                                os.path.basename(fuzzer),
+                                                src_in_docker)
+
+    if covered_files and set(covered_files).intersection(set(files_changed)):
```

Comment:
I don't think we should skip fuzzers which we don't have coverage for. I'd write this as follows:

```python
if not covered files:
  # Assume a fuzzer is affected if we can't get its coverage from OSS-Fuzz.
  affected_fuzzers.append(os.path.basename(fuzzer))
  continue

covered_files = set(covered_files)  # Are you sure there might be duplicates, btw? I doubt it.
for file in files_changed:
  if file in covered_files:
      affected_fuzzers.append(os.path.basename(fuzzer))
      continue
```

Reply:
I agree about keeping fuzzers without coverage reports. I was wondering why you chose the iteration comparison rather than set comparison though?

Reply:
using set comparison is redundant, because
a) creating a set isn't necessary
b) intersection iterates through all elements, while it's sufficient to find just one common element

Reply:
hm I think I've just posted a reply to this but I don't see it, let me repeat just in case: set comparison is redundant because:
a) we don't really need to create sets here
b) intersection iterates through all elements, while we need to find at least one common across two lists

Reply:
Makes sense. Fixed.